File: my-app/src/api/models/course.ts
import { User } from "./user";

export interface Course {
  id: string;
  title: string;
  description: string;
  category: string;
  teacherName: string;
  creationDate: Date;
  lastModified: Date;
  isAdded?: boolean;
}


export interface UserCourse {
  user: User
  appUserId: string
  course: Course
  courseId: string
}

File: API/Models/Course.cs
namespace API.Models;

public class Course
{
  public Guid Id { get; set; }
  public string Title { get; set; }
  public string Description { get; set; }
  public DateTime CreationDate { get; set; }
  public DateTime LastModified { get; set; }
  public string Category { get; set; }
  public string TeacherName { get; set; }
}
